packages/headless/doc-parser/src/entity-sorter.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  AnyEntity,
+  Entity,
+  EntityWithTypeAlias,
+  FuncEntity,
+  ObjEntity,
+} from './entity';
+
+export function sortEntities(entities: AnyEntity[]) {
+  const [methods, attributes] = divideIntoGroups(entities, (entity) =>
+    entity.kind === 'function' ? SortGroup.First : SortGroup.Second
+  ) as [FuncEntity[], (Entity | EntityWithTypeAlias | ObjEntity)[]];
```

Comment:
We should try to avoid casts when possible because they cause us to lose the benefits of typescript. e.g. if someone inverts the order, typescript will not catch the mistake.

We should either correct the return type of the `divideIntoGroups` function, or use separate loops with the `isFunctionEntity` type-guard from which typescript can auto-infer the return type. I think the latter will be simpler with no noticeable hit to performance,